traits/src/crypto.rs

```diff
@@ -113,6 +113,50 @@ pub trait OpenMlsCrypto {
         aad: &[u8],
     ) -> Result<Vec<u8>, CryptoError>;
 
+    /// HPKE single-shot setup of a sender and immediate export a secret.
+    ///
+    /// For the base and PSK modes this encapsulates the public key `pk_r`
+    /// of the receiver.
+    /// For the Auth and AuthPSK modes this encapsulates and authenticates
+    /// the public key `pk_r` of the receiver with the senders secret key `sk_s`.
+    ///
+    /// The encapsulated secret is returned together with the exported secret.
+    /// If the secret key is missing in an authenticated mode, an error is
+    /// returned.
+    fn hpke_setup_sender_and_export(
+        &self,
+        config: HpkeConfig,
+        pk_r: &[u8],
+        info: &[u8],
+        psk: Option<&[u8]>,
+        psk_id: Option<&[u8]>,
+        sk_s: Option<&[u8]>,
+        exporter_context: &[u8],
+        exporter_length: usize,
+    ) -> Result<(Vec<u8>, Vec<u8>), CryptoError>;
```

Comment:
The return values could get an alias or something to distinguish which is which.